File: Ruby/Endpoint Examples/Multipart Payload/translated-pdf-text.rb
#!
# Translate PDF text via single multipart/form-data request.

require "faraday"
require "faraday/retry"
require "faraday/multipart"
require "dotenv"

Dotenv.load

API_KEY = ENV["PDFREST_API_KEY"]
abort("Missing PDFREST_API_KEY in .env") if API_KEY.nil? || API_KEY.strip.empty?
API_BASE = (ENV["PDFREST_URL"] || ENV["PDFREST_API"] || "https://api.pdfrest.com").sub(%r{/+$}, "")

pdf_path = ARGV[0]
abort("Usage: ruby translated-pdf-text.rb /path/to/file.pdf") unless pdf_path && File.file?(pdf_path)

filename = File.basename(pdf_path)

begin
  conn = Faraday.new(url: API_BASE) do |f|
    f.request :retry, max: 2, interval: 0.2
    f.request :multipart
    f.adapter Faraday.default_adapter
  end
  body = {
    file: Faraday::Multipart::FilePart.new(pdf_path, "application/pdf", filename),
    output_language: "en-US", # Translates text to American English. Format the output_language as a 2-3 character ISO 639 code, optionally with a region/script (e.g., 'en', 'es', 'zh-Hant', 'eng-US').
  }
  resp = conn.post("/translated-pdf-text") do |req|
    req.headers["api-key"] = API_KEY
    req.body = body
  end
  puts resp.body
  abort("Translate failed with status #{resp.status}") unless resp.success?
rescue => e
  abort("Error: #{e.class}: #{e.message}")
end

File: Perl/Endpoint Examples/Multipart Payload/translated-pdf-text.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;
use FindBin qw($Bin);
use File::Basename qw(basename);
use LWP::UserAgent;
use HTTP::Request::Common qw(POST);
use Dotenv;

binmode STDOUT, ':raw';
binmode STDERR, ':encoding(UTF-8)';

my $env_path = "$Bin/../../.env";
-e $env_path and Dotenv->load($env_path);

my $api_key = $ENV{PDFREST_API_KEY} // '';
if (!$api_key || $api_key =~ /^\s*$/) { print STDERR "Missing PDFREST_API_KEY\n"; exit 1; }

my $api_base = $ENV{PDFREST_URL} // $ENV{PDFREST_API} // 'https://api.pdfrest.com';
$api_base =~ s{/+$}{};

my $pdf_path = shift @ARGV;
if (!$pdf_path || !-f $pdf_path) { print STDERR "Usage: perl translated-pdf-text.pl /path/to/file.pdf\n"; exit 1; }

my $filename = basename($pdf_path);
my $ua = LWP::UserAgent->new( timeout => 60 );

eval {
    my $req = POST("$api_base/translated-pdf-text",
        'Content_Type' => 'form-data',
        'Content' => [
            file => [$pdf_path, $filename, 'Content-Type' => 'application/pdf'],
            output_language => 'en-US', # Translates text to American English. Format the output_language as a 2-3 character ISO 639 code, optionally with a region/script (e.g., 'en', 'es', 'zh-Hant', 'eng-US').
        ]
    );
    $req->header('api-key' => $api_key);
    my $resp = $ua->request($req);
    print STDOUT $resp->decoded_content // '';
    if (!$resp->is_success) { print STDERR "\nTranslate failed with status " . $resp->code . "\n"; exit 1; }
    1;
} or do { my $err = $@ || 'Unknown error'; $err =~ s/\s+$//; print STDERR "Error: $err\n"; exit 1; };

__END__

